Wallet: Disable -fallbackfee by default

jtimon opened the pull request:
Before it was 0 by default for main and 20000 for test and regtest.
Now it is 0 by default for all chains, thus there's no need to call Params().

Also now the default for main is properly documented.

Suggestion for release notes:

-fallbackfee was 0 (disabled) by default for the main chain, but 20000 by default for the test chains. Now it is 0 by default for all chains. Testnet and regtest users will have to add fallbackfee=20000 to their configuration if they weren't setting it and they want it to keep working like before.

Should I propose them to the wiki for the release notes or only after merge?

For more context, see https://github.com/bitcoin/bitcoin/pull/16402#issuecomment-515701042

jtimon committed:
Truly decouple wallet from chainparams for -fallbackfee

Before it was 0 by default for main and 20000 for test and regtest.
Now it is 0 by default for all chains, thus there's no need to call Params().

Also now the default for main is properly documented

MarcoFalke commented:
Fine with me, but I doubt the tests are passing with this change.

jtimon commented:
> Fine with me, but I doubt the tests are passing with this change.

Oh, right, the tests that make use of fallbackfee will need to set fallbackfee=20000 explicitly now, I'm happy to fix that, but please tell me if I miss any of the tests, specially feature_pruning,feature_dbcrash,feature_fee_estimation which seem to be the tests I forget about the most lately for some reason.

Extra thanks for the fast feedback.

jtimon commented:
Added 1 commit to document current failures and one failed attempt to fix all of them at once.
I'm not going to continue for todat, sorry,

promag commented:
Concept ACK. For now fallback fee could be set for all tests IMO, unless setting explicitly in each tests results in a short change.

MarcoFalke commented on test/functional/test_framework/util.py:
```diff
@@ -295,6 +295,8 @@ def initialize_datadir(dirname, n):
         f.write("[regtest]\n")
         f.write("port=" + str(p2p_port(n)) + "\n")
         f.write("rpcport=" + str(rpc_port(n)) + "\n")
+        # TODO: Do it per test or in the test framework, don't be lazzy
```
I think this is fine to do here.

```suggestion
```

jtimon commented:
Alright, setting it up for all tests is easier, I guess, not a strong opinion.
Removed the rebased, squashed and removed the TODOs. 
Appveyor failed previously on feature_fee_estimation but it's now passing locally, not sure what happened there.

laanwj commented:
Concept ACK

MarcoFalke commented:
ACK 47c0246c44b687f369c15e885c9ef071c22a2597

MarcoFalke commented:
Can you add a file `./doc/release-notes-16524.md` with the content:

```md
Low-level changes
=================

Tests
---

- `-fallbackfee` was 0 (disabled) by default for the main chain, but 20000 by default for the test chains. Now it is 0 by default for all chains. Testnet and regtest users will have to add fallbackfee=20000 to their configuration if they weren't setting it and they want it to keep working like before. (#16524)

jtimon commented:
Fixed nit, added the suggested release notes file.

As a reminder, this is incompatible with https://github.com/bitcoin/bitcoin/pull/16771 and we don't want to merge both of them.
Personally, I would be happy with either one of them.

MarcoFalke commented:
ACK ea4cc3a7b36a9c77dbf0aff439da3ef0ea58e6e4

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK ea4cc3a7b36a9c77dbf0aff439da3ef0ea58e6e4
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgysQwAhkIYBFGteQsfMxLROVL2n3C3HXXl/sufCxtEeyZUTkZJBAV+6Qu/lRc7
8eaFGekk92tvnoMIKQL5pANIJfxWKJQsSOKtZGAa+zQ3CtAvZJwiZPZOMpUPorz2
MeyaBVf4UrmrDdq1L4gJMv0pfE7RsSG1Svl6weDm03Nb2C81inZwzqsLkrqNAaCj
TauZPi+mad2wX7SS2r0VHIp4/ZFsHijJbLoTd30f0q0GJQXnoV9zR8LuAvZ//Fu0
wnUaawJ2yO4YeiGY0K0OGoI+LE4zGzXuRcMa1UFRZ5xGtJfoHQLt1fk4kI8FbNPl
63KQJIIXyYrERlv78NJpzhWK1IfeO4RvLakw/RlnXwKmpSpjIBdKI5Ugj+AI74e2
N+CKrlCZw8b+c7MzTu0ABRipYuF/V9xgjuLPkakqyHsCALDVAS0Kd4ZkPdi7qOPL
Jk82Tit07J5jBYYu6xGO8eCvwMpjINYwG+vgEe9CAUu8hfh7wq6QG4WSaTpJ+nJX
TtgZij77
=SeZE
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `0fc571c1687439a4b0b402de2f3462f3b403a6ae5f6c7f06e547b017fe116a3d  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401080fc571c1687439a4b0b402de2f3462f3b403a6ae5f6c7f06e547b017fe116a3df010357bbe0774323bfcbeb7d8a838d3377d08fff010207caee6a1c0672e9ca83db70a70e1ca08f1045d94e181f008f8d712e76a2155720083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0104094d35507fb9c85bc3ec6c4741463d308f02096e9d53db8c3ac63e5aa75da0a32581cfea0c824d860648385f47285dfc04aa808f1045d94e182f008e6c847d5028a88b60083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010797fbb6bd2c4b66aee48ee9f63bf086d08f1045d94e181f008b35fe53a81186ca70083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010b44a016b0678d697ae6b80571ba5fefe08f020d75cad496791342725aebbec88dd62295d1f00ac6c6db45e4fe19d35d2c892b208f1045d94e182f008a28de777ba506ad30083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

jtimon commented:
Oops, actually the release notes are wrong. People have to set fallbackfee=0.0002 rather than fallbackfee=20000 because it is in BTC, not satoshis.
What to do? a PR only for that?
